src/EventStore.Core/TransactionLog/Scavenging/Sqlite/SqliteScavengeBackend.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using EventStore.Common.Log;
+using EventStore.Common.Utils;
+using Microsoft.Data.Sqlite;
+
+namespace EventStore.Core.TransactionLog.Scavenging.Sqlite {
+	public class SqliteScavengeBackend {
+		protected static readonly ILogger Log = LogManager.GetLoggerFor<SqliteScavengeBackend>();
+	}
+
+	// Encapsulates a connection to sqlite, complete with prepared statements and an API to access it
+	public class SqliteScavengeBackend<TStreamId> : SqliteScavengeBackend, IScavengeStateBackend<TStreamId> {
+		// WAL with SYNCHRONOUS NORMAL means that
+		//  - commiting a transaction does not wait to it to flush to disk
+		//  - which is nice and quick, but means in powerloss the last x transactions
+		//    can be lost. the database will be in a valid state.
+		//  - this is suitable for us because scavenge will continue from the last
+		//    persisted checkpoint.
+		private const string SqliteWalJournalMode = "wal";
+		private const int SqliteNormalSynchronousValue = 1;
+		private const int DefaultSqliteCacheSize = 2 * 1024 * 1024;
+		private const int SqlitePageSize = 16 * 1024;
```

Comment:
is there a specific reason for choosing a page size of 16k vs the default (4k) ? it will result in a database that's 4 times larger than the default. most of the rows we insert appear small enough to very easily fit into 4kb with a lot of spare space left.

Reply:
I ran some tests and larger page size resulted in better performance.
```
Page size of 4k: Read all active records from checkpoint in: 00:00:27.4874806 with 50,000,000 archived & 1,000,000 active, inserted in 00:14:01.5503100.
Page size of 8k: Read all active records from checkpoint in: 00:00:20.5790170 with 50,000,000 archived & 1,000,000 active, inserted in 00:13:10.9948420.
Page size of 16k: Read all active records from checkpoint in: 00:00:14.4001665 with 50,000,000 archived & 1,000,000 active, inserted in 00:12:36.9723471.
Page size of 32k: Read all active records from checkpoint in: 00:00:12.8561650 with 50,000,000 archived & 1,000,000 active, inserted in 00:12:01.5615855.
```

Why you think it will cause more io and larger db?
I would think it's less io but haven't measured it, will see if I can get sum numbers on this.

Reply:
Regarding the size of the database, it gets smaller with larger page size I assume because of less overhead.

Reply:
i'll give it a try with in a real scavenge, the larger page size would be good for reading through all the records but bad for particular lookups, and scavenge does both

Reply:
@pvanbuijtene it was by bad - i incorrectly assumed that only one record was stored per b-tree page while reading the docs: https://www.sqlite.org/fileformat.html but it says: "A leaf page contains _keys_ and in the case of a table b-tree each key has associated data."

Reply:
Closing this. Tim did some tests with different page sizes and real data. 16k is slightly better